### PR TITLE
Live start time tweaks

### PR DESCRIPTION
**Pull request by bc-bbay**
Moving live start time calculation earlier in the code flow to eliminate a race condition, repeating it on first play to create a smooth experience, and making sure currentTime is updated to match these changes.

### CODE REVIEW

File: src/videojs-hls.js [hunk 1]
```diff
@@ -239,7 +244,14 @@ videojs.Hls.prototype.src = function(src) {
   });
 };
 
-videojs.Hls.setMediaIndexForLive = function(selectedPlaylist) {
+/* Returns the media index for the live point in the current playlist, and updates
+   the current time to go along with it.
+ */
+videojs.Hls.getMediaIndexForLive = function(selectedPlaylist) {
+  if (!selectedPlaylist.segments) {
```
Comment (dmlap) [inline src/videojs-hls.js]: Should we be calling this function with a half-loaded playlist now?

File: src/videojs-hls.js [hunk 2]
```diff
@@ -279,6 +291,11 @@ videojs.Hls.prototype.play = function() {
     this.mediaIndex = 0;
   }
 
+  if (this.playlists.media() && !this.el().classList.contains('vjs-has-started')) {
```
Comment (dmlap) [inline src/videojs-hls.js]: We don't support IE8 in this project but you should get into the habit of using `this.hasClass()` for maximum compatibility.

Comment (gkatsev) [inline src/videojs-hls.js]: I think classList actually exists in all the browsers we care to support by this project.

File: src/videojs-hls.js [hunk 3]
```diff
@@ -973,6 +990,20 @@ videojs.Hls.getMediaIndexByTime = function(playlist, time) {
   return -1;
 };
 
+videojs.Hls.prototype.setCurrentTimeByMediaIndex = function(playlist, mediaIndex) {
```
Comment (dmlap) [inline src/videojs-hls.js]: Please document this function. Also, we should denote it as private (append an `_`) unless there's a good reason it should be exposed.

Comment (dmlap) [inline src/videojs-hls.js]: Also, "set" in the name implies some sort of mutation to me. I like side-effect free but the name should change.

File: src/videojs-hls.js (hunk 2 above)
Comment (dmlap) [inline src/videojs-hls.js]: For consistency with the rest of video.js, I think it's still better to use hasClass(). Plus, it's actually less typing here.

File: src/videojs-hls.js (hunk 1 above)
Comment (bc-bbay) [inline src/videojs-hls.js]: Never in production, in theory, but we do it all the time in the unit tests. This is the breakage I was seeing after my first two commits.

File: src/videojs-hls.js (hunk 2 above)
Comment (gkatsev) [inline src/videojs-hls.js]: Yep, consistency is good.
I'd love to move everything to classList, though.

Comment (bc-bbay) [inline src/videojs-hls.js]: Will update, thanks.

File: test/videojs-hls_test.js [hunk 4]
```diff
@@ -1720,6 +1739,11 @@ test('calling play() at the end of a video resets the media index', function() {
   player.hls.ended = function() {
     return true;
   };
+
+  player.hasClass = function(classname) {
```
Comment (dmlap) [inline test/videojs-hls_test.js]: Why is this necessary?

File: src/videojs-hls.js (hunk 1 above)
Comment (dmlap) [inline src/videojs-hls.js]: Should those tests be invoking this method without segments?

**Comment by dmlap** [comment]
@bc-bbay can you append an `_` to the methods added in this commit? I want to be clear they're not intended for external use.

File: src/videojs-hls.js (hunk 1 above)
Comment (bc-bbay) [inline src/videojs-hls.js]: Yeah, in some cases, that's the point.

File: test/videojs-hls_test.js (hunk 4 above)
Comment (bc-bbay) [inline test/videojs-hls_test.js]: hasClass isn't defined in this scope, but you recommended using it in a previous PR review, so this addition keeps the test working.

**Comment by bc-bbay** [comment]
Will do.
